src/legacy/core_plugins/kibana/index.js

```diff
@@ -54,6 +54,8 @@ export default function (kibana) {
         defaultAppId: Joi.string().default('home'),
         index: Joi.string().default('.kibana'),
         disableWelcomeScreen: Joi.boolean().default(false),
+        autocompleteTerminateAfter: Joi.number().default(10000),
+        autocompleteTimeout: Joi.string().default('1s'),
```

Comment:
By making this a string field, the validation for [time-units](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#time-units) will not occur by joi and invalid values will be allowed. Lets make it a number instead and make it the number of milliseconds like `elasticsearch.requestTimeout` configuration

Reply:
Funny enough, I actually started that way and changed it because I looked at the implementation of the elasticsearch configuration settings (`src/core/server/elasticsearch/elasticsearch_config.ts`). It's not documented (yet) but, they support both ways (number as milliseconds or an explicit time unit). I changed it to number and left a TODO for now because the schema type `duration` is only available on NP